auth0/src/test/java/com/auth0/android/authentication/storage/CredentialsManagerTest.java

```diff
@@ -553,6 +554,27 @@ public void shouldRecreateTheCredentials() {
         assertThat(credentials.getScope(), is("openid profile"));
     }
 
+    @Test
+    public void shouldUseCustomClock() {
+        CredentialsManager manager = new CredentialsManager(client, storage);
+
+        long expirationTime = CredentialsMock.CURRENT_TIME_MS; //Same as current time --> expired
+        when(storage.retrieveLong("com.auth0.expires_at")).thenReturn(expirationTime);
+        when(storage.retrieveLong("com.auth0.cache_expires_at")).thenReturn(expirationTime);
+        when(storage.retrieveString("com.auth0.refresh_token")).thenReturn(null);
+        when(storage.retrieveString("com.auth0.access_token")).thenReturn("accessToken");
+        assertThat(manager.hasValidCredentials(), is(false));
```

Comment:
these tests ensure first that the credentials were deemed expired (due to the date) and when the clock instance is swapped, they get fixed